### PR TITLE
FEAT: Modifying CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by PetJs**
Updating the rag_functions.py file explaining the new functions added. I also included the new files that were added to the code base

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -73,8 +73,10 @@ The **Speech-Enabled RAG Solution** is a voice-powered interface that allows use
 The project is structured as follows:
 - **speech_to_docs**: This is the main directory for the project.
 - **speech_to_docs/src**: This directory contains all the files that will house all the functionalities of the project: Speech transcription and synthesis, RAG model Solution and document reading.
-- **speech_to_docs/src/rag_functions.py**: This file contains the function that checks for uploaded file compatibility, making sure it doesn't exceed 50 pages limit as well as the RAG model solution functionalities.
+- **speech_to_docs/src/rag_functions.py**: This file contains functions for checking the uploaded file compatibility, making sure files do not exceed a 50-page limit. It also includes functionalities for processing various document types (PDF, PPTX, TXT) to extract content using Azure Document Intelligence. It also provides detailed logging for error handling and tracks the extraction process, saving the output in a user-friendly text format.
```
Comment (Sammybams) [inline CONTRIBUTING.md]:
```suggestion
- **speech_to_docs/src/rag_functions.py**: This file contains functions for checking the uploaded file compatibility, making sure files do not exceed a 50-page limit. It also includes functionalities for processing various document types (PDF, PPTX, TXT) to extract content using Azure Document Intelligence. It provides detailed logging for error handling and tracks the extraction process, saving the output in a user-friendly text format.

```

Comment (PetJs) [reply CONTRIBUTING.md]: Ok

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -73,8 +73,11 @@ The **Speech-Enabled RAG Solution** is a voice-powered interface that allows use
...
-- **speech_to_docs/src/rag_functions.py**: This file contains the function that checks for uploaded file compatibility, making sure it doesn't exceed 50 pages limit as well as the RAG model solution functionalities.
+- **speech_to_docs/src/rag_functions.py**: This file contains functions for checking the uploaded file compatibility, making sure files do not exceed a 50-page limit. It also includes functionalities for processing various document types (PDF, PPTX, TXT) to extract content using Azure Document Intelligence. It provides detailed logging for error handling and tracks the extraction process, saving the output in a user-friendly text format.
+
 - **speech_to_docs/src/speech_io.py**: This files handles the speech_to_text/ text_to_speech function of the model by using **Azure Cognitive Services: Speech Transcription** (Speech-to-Text) and **Speech Synthesis** (Text-to-Speech).
+- **speech_to_docs/.env.example**: This files handles the environment variables that are used in your codebase. It outlines the essential environment variables needed for the proper functioning of the Speech-to-Docs application.
```
Comment (Sammybams) [inline CONTRIBUTING.md]:
```suggestion
- **speech_to_docs/.env.example**: This file provides a sample of the environment variables used in the codebase. It outlines the essential environment variables needed proper functioning of the Speak-to-Docs solution.
```

**Review by PetJs — COMMENTED** [review]
Speech or speak?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -73,8 +73,11 @@ The **Speech-Enabled RAG Solution** is a voice-powered interface that allows use
...
+- **speech_to_docs/src/rag_functions.py**: This file contains functions for checking the uploaded file compatibility, making sure files do not exceed a 50-page limit. It also includes functionalities for processing various document types (PDF, PPTX, TXT) to extract content using Azure Document Intelligence. It provides detailed logging for error handling and tracks the extraction process, saving the output in a user-friendly text format.
+
 - **speech_to_docs/src/speech_io.py**: This files handles the speech_to_text/ text_to_speech function of the model by using **Azure Cognitive Services: Speech Transcription** (Speech-to-Text) and **Speech Synthesis** (Text-to-Speech).
+- **speech_to_docs/.env.example**: This files handles the environment variables that are used in your codebase. It outlines the essential environment variables needed for the proper functioning of the Speech-to-Docs application.
+- **speech_to_docs/.gitignore**: This contains all the folder and files you do not want to push to github (e.g. .env, /bin/ e.t.c)
```
Comment (Sammybams) [inline CONTRIBUTING.md]:
```suggestion
- **speech_to_docs/.gitignore**: This contains all the folder and files that are not to be pushed to GitHub (e.g. .env, bin/ e.t.c)

```

**Comment by Sammybams** [comment]
> Speech or speak?

It is `speak`, as indicated in the repo.

**Review by Sammybams — APPROVED** [review]
Thank you for your contribution @PetJs . I have made some grammatical structure changes and explanation of some of the files. You can check the commits to see those changes.

**Comment by PetJs** [comment]
Ok

**Comment by PetJs** [comment]
Thanks

**Pull request merged**